test/runtests.jl

```diff
@@ -1,19 +1,19 @@
-using Base.Test, QuantumInfo
+using QuantumInfo
+using Test
 
-import Base.isapprox
-
-rtoldefault = Base.rtoldefault
+import LinearAlgebra
+import QuantumInfo.eye
 
 #function isapprox(m1::Matrix,m2::Matrix; rtol::Real=rtoldefault(abs(m1[1,1]),abs(m2[1,1])), atol::Real=0)
 #  all(x->isapprox(x,0.0,rtol=rtol,atol=atol),abs(m1-m2))
 #end
 
-function isapprox{T1,T2}(v1::Vector{Matrix{T1}},v2::Vector{Matrix{T2}}; rtol::Real=rtoldefault(abs(v1[1][1,1]),abs(v2[1][1,1])), atol::Real=0)
+function Base.isapprox(v1::Vector, v2::Vector; atol::Real=0, rtol::Real=Base.rtoldefault(abs(v1[1][1,1]),abs(v2[1][1,1]), atol))
```

Comment:
Here again I would prefer the parametric definition.

Reply:
Makes sense.  But, `isapprox`  in `Base` now supports this usage out of the box. I prepared a PR that I haven't yet made that  removes all method definitions from `runtests.jl`. One big advantage is that the complexity and debugging of `isapprox` is handled by other people for all Julia code. If you still want type checking it think a one-line wrapper would do it.

Reply:
What do you think of this ? It's a standard construction.
```
function Base.isapprox(v1::Vector{Matrix{T1}}, v2::Vector{Matrix{T2}};
                       atol::Real=0, rtol::Real=rtoldefault(T1, T2, atol)) where {T1, T2}
    invoke(isapprox, Tuple{AbstractVector, AbstractVector}, v1, v2; atol=atol, rtol=rtol, norm=normInf)
end
```
This guarantees that `v1` and `v2` have uniform element types. 

It uses `normInf`, which is what was hand coded before, and is more restrictive than the default 2-norm. 

It makes Julia do most of the work.

Reply:
I didn't realize that `isapprox` in `Base` already supports this. I'm not wedded to using `normInf` here, so perhaps we should just rely on the `Base` definition.